dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/dataset/DatasetCitationsCountUpdater.java

```diff
@@ -0,0 +1,107 @@
+package edu.harvard.iq.dataverse.dataset;
+
+import com.google.common.base.Stopwatch;
+import edu.harvard.iq.dataverse.globalid.DataCiteFindDoiResponse;
+import edu.harvard.iq.dataverse.globalid.DataCiteRestApiClient;
+import edu.harvard.iq.dataverse.persistence.GlobalId;
+import edu.harvard.iq.dataverse.persistence.dataset.Dataset;
+import edu.harvard.iq.dataverse.persistence.dataset.DatasetCitationsCount;
+import edu.harvard.iq.dataverse.persistence.dataset.DatasetCitationsCountRepository;
+import edu.harvard.iq.dataverse.persistence.dataset.DatasetRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.ejb.EJB;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Service responsible for creating/updating {@link DatasetCitationsCount}s
+ * entries in database.
+ * 
+ * @author madryk
+ */
+@ApplicationScoped
+public class DatasetCitationsCountUpdater {
+    private static final Logger logger = LoggerFactory.getLogger(DatasetCitationsCountUpdater.class);
+    
+    @EJB
+    private DatasetRepository datasetRepository;
+    @EJB
+    private DatasetCitationsCountRepository datasetCitationsCountRepository;
+    @Inject
+    private DataCiteRestApiClient dataCiteRestApiClient;
```

Comment:
I think we should continue our way of injecting through constructor.

Reply:
ok